walnuts_cpp/CMakeLists.txt

```diff
@@ -143,3 +147,29 @@ if (CLANG_FORMAT_BIN)
 else()
   message(WARNING "clang-format not found. 'format' target will not be available.")
 endif()
+
+##########################
+##     Doxygen Docs     ##
+##########################
+#################################
+##        Documentation        ##
+#################################
+find_package(Doxygen REQUIRED)
+
+# Public headers only
+file(GLOB_RECURSE WALNUT_HEADERS
+     "${CMAKE_CURRENT_SOURCE_DIR}/include/walnuts/*.hpp")
+
+# Doxygen configuration overrides
+set(DOXYGEN_OUTPUT_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/doc")
```

Comment:
I don't think we should have this output into the source directory, personally

Reply:
To try to avoid all the confusion we've been having, I'm trying to stick to "client" for users of the code and "developers" for people who are building it.  When you say "user's" here, do you mean developers?  I don't think we should expect clients to download Doxygen and CMake to build our docs, so I can't imagine you mean clients.

Reply:
I don't believe I have used the word clients in this thread. Really all I am saying is I think we should delete this `set` line -- the default cmake picks for us is more conventional.

If a user wants to build our docs themselves, they certainly will need cmake and doxygen, but I'm not sure that is relevant to this discussion.